src/SurvivalEstimator.jl

```diff
@@ -338,6 +338,128 @@ function StatsBase.fit!(obj::KaplanMeier, Y::RCSurv)
     )
 end
 
+#-------------------
+# Log-rank tests
+#-------------------
+function update_wt(wt, nevents, nrisk, wtmethod)
+    if wtmethod == :logrank
+        return 1.0
+    elseif wtmethod == :wilcoxon
+        return nrisk
+    elseif wtmethod == :tw
+       return sqrt(nrisk)
+    elseif wtmethod == :peto
+        return wt * (1 - nevents / (nrisk + 1))
+    else
+        error("Unknown weight")
+    end
+end
+
+# Helper function to calculate the score vector u and covariance matrix V for a logrank test.
+function logrank_moments(Y::RCSurv...; wtmethod::Symbol=:logrank)
+    m = length(Y)
+    A = merge(Y...)
+    ti = unique_outcome_times(A)
+    sta = A.stats
+    st = [_padstats(y.stats, ti) for y in Y]
+
+    u = zeros(m)
+    V = zeros(m, m)
+    wt = 1.0
+    for i in eachindex(ti)
+        d, n = sta.nevents[i], sta.nrisk[i]
+        wt = update_wt(wt, d, n, wtmethod)
+        if n == 0
+            break
+        end
+        for j in 1:m
+            dd, nnj = st[j].nevents[i], st[j].nrisk[i]
+            rj = dd / nnj
+            fj = nnj / n
+            u[j] += wt * (dd - d*fj)
+            for k in 1:m
+                nnk = st[k].nrisk[i]
+                fk = nnk / n
+                q = j == k ? 1.0 : 0.0
+                if n > 1
+                    V[j,k] += wt^2 * (q - fj) * fk * d * (n - d) / (n - 1)
+                end
+            end
+        end
+    end
+
+    return u, V
+end
+
+"""
+    logrank_test(Y::RCSurv...; wtmethod=:logrank)
+
+Test the null hypothesis that two or more survival functions are identical.
+"""
```

Comment:
Please make this more informative, see other documentation. Include examples and make clear what the choices are for `wtmethod`